File: src/main/java/com/hyecheon/antteshop/entity/Role.java
package com.hyecheon.antteshop.entity;

import lombok.*;

import javax.persistence.*;

/**
 * User: hyecheon lee
 * Email: rainbow880616@gmail.com
 * Date: 2021/08/14
 */

@EqualsAndHashCode(callSuper = true)
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "roles")
public class Role extends BaseWithUserEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(length = 40, nullable = false, unique = true)
    private String name;

    @Column(length = 150, nullable = false)
    private String description;
}

File: src/main/kotlin/com/hyecheon/antteshop/domains/entity/Role.kt
package com.hyecheon.antteshop.domains.entity

import com.hyecheon.antteshop.entity.BaseWithUserEntity
import lombok.*
import javax.persistence.*

/**
 * User: hyecheon lee
 * Email: rainbow880616@gmail.com
 * Date: 2021/08/14
 */
@Entity
@Table(name = "roles")
data class Role(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(length = 40, nullable = false, unique = true)
    var name: String? = null,

    @Column(length = 150, nullable = false)
    var description: String? = null,
) : BaseWithUserEntity()